From de3886c215e54659e9e219243dc7aa17d00507e0 Mon Sep 17 00:00:00 2001
From: Gary Lin <glin@suse.com>
Date: Tue, 3 Dec 2019 14:00:40 +0800
Subject: [PATCH] Cryptlib: update for efi build

This patch is split from Cryptlib.diff. Instead of covering the whole
Cryptlib, it focuses on the building issues of the wrappers in Syscall.

* SysCall/CrtWrapper.c: Remove the dummy implementation of BIO_printf()
  and BIO_snprintf() since MokManager would need BIO_snprintf() to
  extract the information of a certificate

* SysCall/TimerWrapper.c: Remove the headers we don't need

Signed-off-by: Gary Lin <glin@suse.com>
---
 Cryptlib/SysCall/CrtWrapper.c   | 14 --------------
 Cryptlib/SysCall/TimerWrapper.c |  2 --
 2 files changed, 16 deletions(-)

diff --git a/Cryptlib/SysCall/CrtWrapper.c b/Cryptlib/SysCall/CrtWrapper.c
index 42235ab..1ee0e32 100644
--- a/Cryptlib/SysCall/CrtWrapper.c
+++ b/Cryptlib/SysCall/CrtWrapper.c
@@ -383,20 +383,6 @@ size_t fwrite (const void *buffer, size_t size, size_t count, FILE *stream)
   return 0;
 }
 
-//
-//  -- Dummy OpenSSL Support Routines --
-//
-
-int BIO_printf (void *bio, const char *format, ...)
-{
-  return 0;
-}
-
-int BIO_snprintf(char *buf, size_t n, const char *format, ...)
-{
-  return 0;
-}
-
 #ifdef __GNUC__
 
 typedef
diff --git a/Cryptlib/SysCall/TimerWrapper.c b/Cryptlib/SysCall/TimerWrapper.c
index 027e5a9..bf4016d 100644
--- a/Cryptlib/SysCall/TimerWrapper.c
+++ b/Cryptlib/SysCall/TimerWrapper.c
@@ -7,9 +7,7 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 **/
 
-#include <Uefi.h>
 #include <CrtLibSupport.h>
-#include <Library/UefiRuntimeServicesTableLib.h>
 
 //
 // -- Time Management Routines --
-- 
2.24.0

